static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -28,68 +28,112 @@ define([
-
-                            fastdom.write(function () {
-                                // Reset
-                                $header.css('transition', '');
-                                $adBanner.css('top', '');
-
-                                // Set
-                                if (scrollY > headerHeight) {
-                                    $adBanner.css({
-                                        'position': 'absolute',
-                                        'top': headerHeight + 'px'
-                                    });
-                                } else {
-                                    $adBanner.css({ 'position': 'fixed' });
-                                }
-
-                                $header.css('margin-top', adHeight + 'px');
-
-                                var scrollIsAtTop = scrollY === 0;
-                                // Avoid an initial transition when we apply the margin top for the first time
-                                if (scrollIsAtTop && !firstRender) {
-                                    // If the user is at the top of the page, we want to transition
-                                    // the change
-                                    $header.css('transition', 'margin-top 0.75s cubic-bezier(0, 0, 0, 0.985)');
-                                } else {
-                                    // If the user is not at the top, we want to offset their scroll position
-                                    var diff = adHeight - previousAdHeight;
-                                    if (diff > 0) {
-                                        window.scrollTo(scrollX, scrollY + diff);
-                                    }
-                                }
+                var getLatestAdHeight = function () {
+                    var $iframe = $('iframe', $adBanner);
+                    var slotWidth = $iframe.attr('width');
+                    var slotHeight = $iframe.attr('height');
+                    // iframe may not have been injected at this point
+                    var isFluidAd = $iframe.length > 0 && [slotWidth, slotHeight].join(',') === '88,70';
+                    // fluid ads are currently always 250px high. We can't just read the client height as fluid ads are
+                    // injected asynchronously, so we can't be sure when they will be in the dom
+                    var fluidAdInnerHeight = 250;
+                    var fluidAdPadding = 18;
+                    var fluidAdHeight = fluidAdInnerHeight + fluidAdPadding;
 
-                                previousAdHeight = adHeight;
+                    return isFluidAd
+                        ? Promise.resolve(fluidAdHeight)
+                        : getClientAdHeight();
+                };
+
+                var oldAdHeightPromise = getLatestAdHeight();
+                var newAdHeightPromise = topAdRenderedPromise.then(getLatestAdHeight);
+
+                var getCachedAdHeight = function () {
```

Comment:
i wonder if it's more immediately comprehensible to name things by their intent or role than what they literally are (which is what i think is happening in the scala convention)? 

in your example, you'll probably be doing either `getUser` or `createUser` or some such. you only end up shadowing `user` because the promise was badly named (i agree). 

the first instance of `user` there is not interesting because it's a `Promise` (specifically), but because it hasn't yet happened.

if i invented an amazing thing called a `Probably`, and we replaced `Promise`, `getUser` and `createUser` would still be meaningful because they only capture the behaviour, not the implementation